tests/kv679_dataloss.erl

```diff
@@ -0,0 +1,161 @@
+%% -------------------------------------------------------------------
+%%
+%% Copyright (c) 2014 Basho Technologies, Inc.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+%%% @copyright (C) 2014, Basho Technologies
+%%% @doc
+%%% riak_test for kv679 lost clock flavour.
+%%%
+%%% issue kv679 is a possible dataloss issue, it's basically caused by
+%%% the fact that per key logical clocks can go backwards in time in
+%%% certain situations. The situation under test here is as follows:
+%%% Create value, write N times
+%%% Fail to locally read value on coordinate (corruption, error, solar flare)
+%%% write new value (new value, new clock!)
+%%% replicate new value
+%%% replicas drop write as clock is dominated
+%%% read repair removes value. Data loss.
+%%% @end
+
+-module(kv679_dataloss).
+-behavior(riak_test).
+-compile([export_all]).
+-export([confirm/0]).
+
+-include_lib("eunit/include/eunit.hrl").
+
+-define(BUCKET, <<"kv679">>).
+-define(KEY, <<"test">>).
+
+confirm() ->
+    Conf = [
+            {riak_kv, [{anti_entropy, {off, []}}]},
+            {riak_core, [{default_bucket_props, [{allow_mult, true},
+                                                 {dvv_enabled, true}]}]},
+            {bitcask, [{sync_strategy, o_sync}, {io_mode, nif}]}],
+
+    [Node] = rt:deploy_nodes(1, Conf),
+    Client = rt:pbc(Node),
+    riakc_pb_socket:set_options(Client, [queue_if_disconnected]),
+
+    %% Get preflist for key
+    %% assuming that the head of the preflist on a single node cluster
+    %% will always coordinate the writes
+    PL = kv679_tombstone:get_preflist(Node),
+
+    %% Write key some times
+    write_key(Client, <<"phil">>, []),
+
+    {ok, Bod} =  write_key(Client, <<"bob">>, [return_body]),
+
+    lager:info("wrote value <<bob>>"),
+
+    VCE0 = riakc_obj:vclock(Bod),
+    VC0 = rpc:call(Node, riak_object, decode_vclock, [VCE0]),
+    lager:info("VC ~p~n", [VC0]),
+
+
+    %% delete the local data for Key
+    %% ERM why not just stop the node, then delete the data dir?
+    delete_datadir(hd(PL)),
+
+    timer:sleep(10000),
```

Comment:
This timeout sticks out as something that may not be helpful for repeatability. Are we sure it's even necessary here?